hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java

```diff
@@ -95,6 +95,19 @@ public HapiFhirJpaMigrationTasks(Set<String> theFlags) {
 		init620();
 		init640();
 		init660();
+		init670();
+	}
+
+	protected void init670() {
+		Builder version = forVersion(VersionEnum.V6_7_0);
+
+		// HAPI-FHIR #4801 - Add New Index On HFJ_RESOURCE
+		Builder.BuilderWithTableName resourceTable = version.onTable("HFJ_RESOURCE");
+		resourceTable
+			.addIndex("20230502.1", "IDX_RESID_UPDATED")
```

Comment:
Minor: It's nice if they all have the same prefix
```suggestion
			.addIndex("20230502.1", "IDX_RES_RESID_UPDATED")
```